### ISSUE TITLE
Handle upgrades that require upgrading multiple dependencies at once

### ISSUE THREAD

**Issue by greysteil** [dependabot/dependabot-core#364]
_From @greysteil on May 31, 2017 10:0_

On a lot of old repos, we see Gemfiles with something like the below:

```
gem "rails", "~> 4.1"
gem "sass-rails", "~> 4.1"
```

To upgrade either of the dependencies in the above Gemfile, we need to update both, because `sass-rails` version 5 depends on the `rails` version being 5.

The right way to handle the above is to put in a PR upgrading both of the above, with clear details of both updates.

_Copied from original issue: dependabot/feedback#5_

**Comment by greysteil** [comment]
_From @gerrywastaken on October 28, 2017 23:10_

For now I'd just be happy with just the addition of a separate PR explaining the issue and listing the gems that are currently blocked from being updated. You can also point out that you hope to address this in the future. However currently this issue is just hidden from most users and they may think they are up to to date when they are actually not.

Btw this is just my idea having no knowledge of how things work currently. I'd be happy with any low tech solution that stops the problem from being hidden.

**Comment by greysteil** [comment]
This is really close now - should have a working implementation of multi-dependency PRs tomorrow. Will only be for cases where multiple dependencies need to be updated at the same time, at least at first (but could be extended to allow people to opt into combining PRs, as mentioned in #52).

**Comment by greysteil** [comment]
_From @gerrywastaken on November 24, 2017 1:10_

Impressive work! I can't wait.

> Will only be for cases where multiple dependencies need to be updated at the same time

Does that still mean some cases would be hidden?

**Comment by greysteil** [comment]
The only case that would be hidden is where an update to the latest version isn't possible at all given your dependencies (e.g., where a new release is required from one of your dependencies to make it compatible with some of your other ones). Creating issues is probably the only way for us to address that case, but I want to see how common it is before committing to that solution.

Basically, Dependabot will be as good as a human at bumping Dependencies once this is done. 🤖

**Comment by greysteil** [comment]
_From @gerrywastaken on November 26, 2017 6:39_

That's awesome. Yeah creating issues would be something that I would certainly like over not having any awareness that certain gems are stuck because another gem is blocking it. This way I can completely depend on dependabot. :)

**Comment by greysteil** [comment]
This is shipped! Example PR [here](https://github.com/greysteil/ruby-test/pull/3).

It needed a *lot* of changes to our architecture, so I'm going to be on standby looking for bugs over the next few days - if you see any, let me know!
